dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/DataManager.kt

```diff
@@ -53,58 +63,152 @@ class DataManager(
+     * * @return stringified data entry from the temporary store
+     */
+    fun selectDataSetFromTemporaryStorage(dataId: String): String {
+        val rawValue = dataInMemoryStorage.getOrElse(dataId) {
+            throw ResourceNotFoundApiException(
+                "Data ID not found in temporary storage",
+                "Dataland does not know the data id $dataId",
+            )
+        }
+        return objectMapper.writeValueAsString(rawValue)
+    }
+
+    /**
+     * Method to temporarily store a data set in a hash map and send a message to the storage_queue
+     * @param storableDataSet The data set to store
+     * @param companyName The name of the company corresponding to the data set to store
+     * @param correlationId The correlation id of the request initiating the storing of data
+     * @return ID of the stored data set
+     */
+    fun storeDataSetInTemporaryStoreAndSendMessage(
         storableDataSet: StorableDataSet,
         companyName: String,
         correlationId: String,
     ): String {
-        val dataId: String
-        try {
-            dataId = storageClient.insertData(correlationId, objectMapper.writeValueAsString(storableDataSet)).dataId
-        } catch (e: ServerException) {
-            val internalMessage = "Error storing data." +
-                " Received ServerException with Message: ${e.message}. Correlation ID: $correlationId"
+        val dataId = generateRandomDataId()
+        dataInMemoryStorage[dataId] = objectMapper.writeValueAsString(storableDataSet)
+        cloudEventMessageHandler.buildCEMessageAndSendToQueue(
+            dataId, "Data received", correlationId,
+            MqConstants.dataReceived,
+        )
+        logger.info(
+            "Stored StorableDataSet of type ${storableDataSet.dataType} for company ID in temporary store" +
+                "${storableDataSet.companyId}. Company Name $companyName received ID $dataId from storage. " +
+                "Correlation ID: $correlationId.",
+        )
+        return(dataId)
+    }
+
+    /**
+     * Method to generate a random Data ID
+     * @return generated UUID
+     */
+    fun generateRandomDataId(): String {
+        return "${UUID.randomUUID()}:${UUID.randomUUID()}_${UUID.randomUUID()}"
+    }
+
+    /**
+     * Method that listens to the stored queue and removes data entries from the temporary storage once they have been
+     * stored in the persisted database. Further it logs success notification associated containing dataId and
+     * correlationId
+     * @param message Message retrieved from stored_queue
+     */
+    @RabbitListener(
+        bindings = [
+            QueueBinding(
+                value = Queue("dataQualityAssuredBackendDataManager"),
+                exchange = Exchange(MqConstants.dataQualityAssured, declare = "false"),
+                key = [""],
+            ),
+        ],
+    )
+    fun removeStoredItemFromTemporaryStore(message: Message) {
```

Comment:
This method listens to the wrong exchange.

Reply:
Corrected, both listeners got confused.